spiketools/spatial/occupancy.py

```diff
@@ -170,121 +140,272 @@ def compute_bin_assignment(position, x_edges, y_edges=None, include_edge=True):
-    >>> compute_bin_time(timestamp)
-    array([1., 2., 3., 2., 1., 0.])
+        # Spatially bin 2d position data, and collect bin assignment information
+        x_edges, y_edges = compute_bin_edges(position, bins, area_range)
+        x_bins, y_bins = compute_bin_assignment(position, x_edges, y_edges)
+        data_dict['xbin'] = pd.Categorical(\
+            x_bins, categories=list(range(0, bins[0])), ordered=True)
+        data_dict['ybin'] = pd.Categorical(\
+            y_bins, categories=list(range(0, bins[1])), ordered=True)
+
+    bindf = pd.DataFrame(data_dict)
+
+    if time_threshold is not None:
+        bindf = bindf[bindf.time < time_threshold]
+
+    if speed_threshold is not None:
+        bindf = bindf[bindf.speed > speed_threshold]
+
+    return bindf
+
+
+def compute_occupancy_df(bindf, bins, minimum=None, normalize=False, set_nan=False):
+    """Compute the bin occupancy from bin-position dataframe.
+
+    Parameters
+    ----------
+    bindf : pd.DataFrame
+        Dataframe representation of position bin information.
+    bins : int or list of [int, int]
+        The bin definition for dividing up the space. If 1d, can be integer.
+        If 2d should be a list, defined as [number of x_bins, number of y_bins].
+    minimum : float, optional
+        The minimum required occupancy.
+        If defined, any values below this are set to zero.
+    normalize : bool, optional, default: False
+        Whether to normalize occupancy to sum to 1.
+    set_nan : bool, optional, default: False
+        Whether to set zero occupancy locations as NaN.
+
+    Returns
+    -------
+    occupancy : 1d or 2d array
+        Computed occupancy across the space.
+        For 2d, has shape [n_y_bins, n_x_bins] (see notes in `compute_occupancy`).
     """
 
-    return np.append(np.diff(timestamps), 0)
+    bins = check_position_bins(bins)
+
+    # Group position samples into spatial bins, summing total time spent there
+    groupby = sorted([el for el in list(bindf.columns) if 'bin' in el])
+    bingroup = bindf.groupby(groupby)['time'].sum()
+    # This standardizes any unobserved values to be zero
+    #   This can otherwise vary by code version (sometimes being NaN instead)
+    bingroup = bingroup.fillna(0)
+
+    # Extract and re-organize occupancy into array
+    occupancy = bingroup.values.reshape(*bins)
+    occupancy = occupancy.T
+
+    if minimum:
+        occupancy[occupancy < minimum] = 0.
+
+    if normalize:
+        occupancy = occupancy / np.sum(occupancy)
+
+    if set_nan:
+        occupancy[occupancy == 0.] = np.nan
+
+    return occupancy
 
 
-def compute_occupancy(position, timestamps, bins, speed=None, speed_thresh=None, minimum=None,
-                      normalize=False, set_nan=False, area_range=None, transpose=True):
+def compute_occupancy(position, timestamps, bins, area_range=None,
+                      speed=None, speed_threshold=None, time_threshold=None,
```

Comment:
`time_threshold` defaults to `None`  here but to `np.inf` in `spiketools/spatial/place/compute_place_bins`. 
Should we set both to the same default?

Reply:
Yeh, good catch! I think it makes more sense (and is computationally a bit better), to make the threshold optional (default to None), and compare if defined, instead of always comparing to inf. I'll update this!